src/js/common/schemaform/definitions/address.js

```diff
@@ -31,6 +31,16 @@ function validateAddress(errors, address, formData, currentSchema) {
     errors.state.addError('Please select a state or province');
   }
 
+  const hasAddressInfo = stateRequiredCountries.has(address.country)
+    && !currentSchema.required.length
+    && typeof address.street !== 'undefined'
+    && typeof address.city !== 'undefined'
```

Comment:
If I add a city and then delete it, it becomes undefined or an empty string?

Reply:
It becomes undefined. Empty strings don't trigger required errors in JSON schema land, so all the form widgets removes data entirely instead of using an empty string.